File: dp_easy_198_不相邻的数求最大和/robber.java
class Solution {
    public int rob(int[] nums) {
        int length=nums.length;
        if(nums==null ||length==0)
        	return 0;
        else if(length==1)
        	return nums[0];
        else
        {
        	int[] f=new int[length];
        	f[0]=nums[0];
        	f[1]=Math.max(f[0],nums[1]);
        	for(int i=2;i<length;i++)
        	{
        		f[i]=Math.max(f[i-1],f[i-2]+nums[i]);
        	}
        	return f[length-1];
        }
    }
}

File: dp_easy_198_不相邻的数求最大和/robber.cpp
int rob(vector<int> &nums)
{
	int size=nums.size();
	if(size==0)
	{
		return NULL;
	}
	else if(size==1)
	{
		return nums[0];
	}
	else
	{
		nums[1]=max(nums[0],nums[1]);
		for(int i=2;i<size;i++)
		{
			nums[i]=max(nums[i-1],nums[i-2]+nums[i]);
		}
		return nums[size-1];

	}
}